django/core/validators.py

```diff
@@ -204,9 +204,9 @@ def __init__(self, message=None, code=None, allowlist=None):
             self.domain_allowlist = allowlist
 
     def __call__(self, value):
-        # The maximum length of an email is 320 characters per RFC 3696
+        # The maximum length of an email is 254 characters per RFC 5321/5322
         # section 3.
-        if not value or "@" not in value or len(value) > 320:
```

Comment:
Tests and documentation changes are also required. Moreover we should do the same in `forms`.

Reply:
I have added tests and updated the forms. The documentation how still requires editing.